tools/analysis/programs.go

```diff
@@ -69,22 +69,53 @@ func (programs Programs) load(
 
 	program, err := parser.ParseProgram(nil, code, parser.Config{})
 	if err != nil {
-		return wrapError(err)
+		// If a parser error handler is set and the error is non-fatal
+		// use handler to handle the error (e.g. to report it and continue analysis)
+		// This will only be used for the entry point program (e.g not an imported program)
+		wrappedErr := wrapError(err)
+		if loadError == nil {
+			loadError = wrappedErr
+		}
+
+		if config.HandleParserError != nil && program != nil && importingLocation == nil {
+			err = config.HandleParserError(wrappedErr)
+			if err != nil {
+				return err
+			}
+		} else {
+			return wrappedErr
+		}
 	}
 
 	var checker *sema.Checker
 	if config.Mode&NeedTypes != 0 {
 		checker, err = programs.check(config, program, location, seenImports)
 		if err != nil {
-			return wrapError(err)
+			wrappedErr := wrapError(err)
+			if loadError == nil {
+				loadError = wrappedErr
+			}
+
+			// If a custom error handler is set, and the error is non-fatal
+			// use handler to handle the error (e.g. to report it and continue analysis)
+			// This will only be used for the entry point program (e.g not an imported program)
+			if config.HandleCheckerError != nil && checker != nil && importingLocation == nil {
```

Comment:
Same as above: Good that there is a comment explaining here when the handler is called, but it might still be surprising to users of the package.